### PR TITLE
MVBASE-1652 | Add Oban tracing

### PR DESCRIPTION
**Pull request by maciej-szlosarczyk**

### CODE REVIEW

**Comment by sauron-droid** [comment]
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/mindvalley/mv-opentelemetry/blob/master/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [86a300c](https://github.com/mindvalley/mv-opentelemetry/commits/86a300cfab49fd4d75eb08cea1ff5b044a159e50) Add Oban tracing

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
**Once this PR has been reviewed and has the lgtm label**, please ask for approval from maciej-szlosarczyk by writing `/assign @maciej-szlosarczyk` in a comment. For more information see [the Kubernetes Code Review Process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process).

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: mix.exs
Comment (maciej-szlosarczyk) [inline mix.exs:47]: This override is no longer needed, from the look of it. It was originally in place to allow for using telemetry 0.4 but this is no longer supported.

File: test/mv_opentelemetry/oban_test.exs
Comment (maciej-szlosarczyk) [inline test/mv_opentelemetry/oban_test.exs:4]: I can't seem to be able to test the ones that return errors, so we will have to trust that it works.

File: lib/mv_opentelemetry/oban.ex
Comment (antonsatin) [inline lib/mv_opentelemetry/oban.ex:15]: 🤔 Whats the difference between this and the default implementation from macro?

**Review by antonsatin — DISMISSED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/mv-opentelemetry/pull/39#pullrequestreview-1224260436" title="Approved">antonsatin</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)~~ [antonsatin]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
LGTM label has been added.  <details>Git tree hash: 98aa4b5309ef64b16eaf171df3a4315f4ebb2f06</details>

File: lib/mv_opentelemetry/oban.ex
Comment (maciej-szlosarczyk) [reply lib/mv_opentelemetry/oban.ex:15]: There was originally a difference, but now there isn't, let me just remove it.

**Comment by sauron-droid** [comment]
New changes are detected. LGTM label has been removed.

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
**Once this PR has been reviewed and has the lgtm label**, please ask for approval from antonsatin by writing `/assign @antonsatin` in a comment. For more information see [the Kubernetes Code Review Process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process).

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
**Once this PR has been reviewed and has the lgtm label**, please ask for approval from antonsatin by writing `/assign @antonsatin` in a comment. For more information see [the Kubernetes Code Review Process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process).

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by onimsha** [comment]
@maciej-szlosarczyk can post an example of how the trace looks like ? On Honeycomb perhaps.

**Comment by sauron-droid** [comment]
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/mindvalley/mv-opentelemetry/blob/master/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [9b6805d](https://github.com/mindvalley/mv-opentelemetry/commits/9b6805d2f956fd5a956c65e091873f9614f028fb) Use worker module name for span name

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by maciej-szlosarczyk** [comment]
@onimsha putting it into Honeycomb is a bit too much, here's a screenshot from Jaeger as well as JSON payload:

Screenshot 2022-12-21 at 10 26 17


```
{
    "data": [
        {
            "traceID": "a6947818104af50fe186a297704c84f4",
            "spans": [
                {
                    "traceID": "a6947818104af50fe186a297704c84f4",
                    "spanID": "26442eacd7cf03e6",
                    "operationName": "MvOpentelemetryHarness.Oban.Job process",
                    "references": [],
                    "startTime": 1671611140336248,
                    "duration": 3966,
                    "tags": [
                        {
                            "key": "messaging.destination",
                            "type": "string",
                            "value": "events"
                        },
                        {
                            "key": "messaging.destination_kind",
                            "type": "string",
                            "value": "queue"
                        },
                        {
                            "key": "messaging.oban.attempt",
                            "type": "int64",
                            "value": 1
                        },
                        {
                            "key": "messaging.oban.inserted_at",
                            "type": "string",
                            "value": "2022-12-21T08:25:40.082660Z"
                        },
                        {
                            "key": "messaging.oban.job_id",
                            "type": "int64",
                            "value": 6
                        },
                        {
                            "key": "messaging.oban.max_attempts",
                            "type": "int64",
                            "value": 1
                        },
                        {
                            "key": "messaging.oban.priority",
                            "type": "int64",
                            "value": 0
                        },
                        {
                            "key": "messaging.oban.scheduled_at",
                            "type": "string",
                            "value": "2022-12-21T08:25:40.082660Z"
                        },
                        {
                            "key": "messaging.oban.worker",
                            "type": "string",
                            "value": "MvOpentelemetryHarness.Oban.Job"
                        },
                        {
                            "key": "messaging.operation",
                            "type": "string",
                            "value": "process"
                        },
                        {
                            "key": "messaging.system",
                            "type": "string",
                            "value": "oban"
                        },
                        {
                            "key": "span.kind",
                            "type": "string",
                            "value": "consumer"
                        },
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "a6947818104af50fe186a297704c84f4",
                    "spanID": "e7a56bd3a3062d4e",
                    "operationName": "inner span",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "a6947818104af50fe186a297704c84f4",
                            "spanID": "26442eacd7cf03e6"
                        }
                    ],
                    "startTime": 1671611140338727,
                    "duration": 17,
                    "tags": [
                        {
                            "key": "span.kind",
                            "type": "string",
                            "value": "internal"
                        },
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "a6947818104af50fe186a297704c84f4",
                    "spanID": "fb4d82375392f6d4",
                    "operationName": "more inner span",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "a6947818104af50fe186a297704c84f4",
                            "spanID": "e7a56bd3a3062d4e"
                        }
                    ],
                    "startTime": 1671611140338735,
                    "duration": 6,
                    "tags": [
                        {
                            "key": "span.kind",
                            "type": "string",
                            "value": "internal"
                        },
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                }
            ],
            "processes": {
                "p1": {
                    "serviceName": "mv-opentelemetry",
                    "tags": [
                        {
                            "key": "process.executable.name",
                            "type": "string",
                            "value": "erl"
                        },
                        {
                            "key": "process.runtime.description",
                            "type": "string",
                            "value": "Erlang/OTP 23 erts-11.1.6"
                        },
                        {
                            "key": "process.runtime.name",
                            "type": "string",
                            "value": "BEAM"
                        },
                        {
                            "key": "process.runtime.version",
                            "type": "string",
                            "value": "11.1.6"
                        },
                        {
                            "key": "service.namespace",
                            "type": "string",
                            "value": "mindvalley"
                        }
                    ]
                }
            },
            "warnings": null
        }
    ],
    "total": 0,
    "limit": 0,
    "offset": 0,
    "errors": null
}
```

**Comment by onimsha** [comment]
@maciej-szlosarczyk from this looks like we're not capturing the `queue_name` from the Oban job's span ? I wonder how could we use this trace for gaining insights about our Job processing system. For more specific, if we want to check which queue has been executed the most and what is the latency of each, how should we do that using the Oban tracing data ?

**Comment by maciej-szlosarczyk** [comment]
@onimsha we do capture queue name as `messaging.destination`. The OTel standard for batch jobs is to use messaging convention, which is the closest, but it's a little bit ill-fitting for now. There's a special standard in the works for batch jobs, we'll switch to that when it becomes more established:

https://github.com/open-telemetry/opentelemetry-specification/issues/1347

**Comment by onimsha** [comment]
> @onimsha we do capture queue name as `messaging.destination`. The OTel standard for batch jobs is to use messaging convention, which is the closest, but it's a little bit ill-fitting for now. There's a special standard in the works for batch jobs, we'll switch to that when it becomes more established:
> 
> [open-telemetry/opentelemetry-specification#1347](https://github.com/open-telemetry/opentelemetry-specification/issues/1347)

cool. Thanks for indulge my nitpicking :) .

**Review by onimsha — DISMISSED** [review]

**Comment by sauron-droid** [comment]
LGTM label has been added.  <details>Git tree hash: 60fa0bf726f9c395284a18fb4eec2fc01d13e620</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/mv-opentelemetry/pull/39#pullrequestreview-1225734131" title="Approved">onimsha</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)~~ [onimsha]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by antonsatin — DISMISSED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/mv-opentelemetry/pull/39#pullrequestreview-1227233776" title="Approved">antonsatin</a>*, *<a href="https://github.com/mindvalley/mv-opentelemetry/pull/39#pullrequestreview-1225734131" title="Approved">onimsha</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)~~ [antonsatin,onimsha]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
@maciej-szlosarczyk: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by sauron-droid** [comment]
New changes are detected. LGTM label has been removed.

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
**Once this PR has been reviewed and has the lgtm label**, please ask for approval from antonsatin by writing `/assign @antonsatin` in a comment. For more information see [the Kubernetes Code Review Process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process).

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
**Once this PR has been reviewed and has the lgtm label**, please ask for approval from antonsatin by writing `/assign @antonsatin` in a comment. For more information see [the Kubernetes Code Review Process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process).

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
**Once this PR has been reviewed and has the lgtm label**, please ask for approval from antonsatin by writing `/assign @antonsatin` in a comment. For more information see [the Kubernetes Code Review Process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process).

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by antonsatin — APPROVED** [review]

**Comment by sauron-droid** [comment]
LGTM label has been added.  <details>Git tree hash: 6a81b0410723b920c568254d229fb830c8d001e1</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/mv-opentelemetry/pull/39#pullrequestreview-1227256243" title="Approved">antonsatin</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)~~ [antonsatin]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
